common/djangoapps/course_modes/models.py

```diff
@@ -583,32 +602,23 @@ def enrollment_mode_display(cls, mode, verification_status):
                 enrollment_value = _("Verified")
                 show_image = True
                 image_alt = _("ID Verified Ribbon/Badge")
-            else:
-                enrollment_title = _("You're enrolled as an honor code student")
-                enrollment_value = _("Honor Code")
-        elif mode == cls.HONOR:
+        elif display_mode == "honor":
             enrollment_title = _("You're enrolled as an honor code student")
             enrollment_value = _("Honor Code")
-        elif mode == cls.AUDIT:
-            enrollment_title = _("You're auditing this course")
-            enrollment_value = _("Auditing")
-        elif mode in [cls.PROFESSIONAL, cls.NO_ID_PROFESSIONAL_MODE]:
+        elif display_mode == "professional":
```

Comment:
sorry for the drive by comment, but why are we dropping no_id_professional here?